### ISSUE TITLE
A Question about Low Correlations in the "stageA2_mbm_finetune" Execution on the GOD Dataset

### ISSUE THREAD

**Issue by HuangWanqiu** [zjc062/mind-vis#26]
Thanks for all your efforts!

When running "stageA2_mbm_finetune" with the GOD dataset, the obtained correlations are very small (approximately 0.02). When I execute "stageA2_mbm_finetune" using the following command, the correlations obtained on the GOD test dataset are extremely low, as shown in the figure below, reaching a maximum of only around 0.02. Is this a normal result?

`python code/stageA2_mbm_finetune.py --dataset GOD --pretrain_mbm_path results/pretrains/GOD/fmri_encoder.pth`

![image](https://github.com/zjc062/mind-vis/assets/12394382/3d58a746-7d9a-4812-8ea7-f6f068e697ee)

**Comment by HuangWanqiu** [comment]
I found the reason. It was my mistake. After increasing the epoch number to 5000 (actually around 3000 would suffice), the correlation on the GOD test set increased to around 0.57, as shown in the figure below.

Once again, thank you for your excellent work.

![image](https://github.com/zjc062/mind-vis/assets/12394382/d4625806-98b8-43fa-ae0d-cd483e78ce3b)

**Comment by HuangWanqiu** [comment]
I have an additional question. In your valuable experience, when adapting the code to another dataset, what would be a "good" value for the correlation (cor) of the train set/test set? 
I look forward to your response.

**Comment by xyhanHIT** [comment]
> I found the reason. It was my mistake. After increasing the epoch number to 5000 (actually around 3000 would suffice), the correlation on the GOD test set increased to around 0.57, as shown in the figure below.
> 
> Once again, thank you for your excellent work.
> 
> ![image](https://private-user-images.githubusercontent.com/12394382/307229897-d4625806-98b8-43fa-ae0d-cd483e78ce3b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3ODI3ODIsIm5iZiI6MTcwOTc4MjQ4MiwicGF0aCI6Ii8xMjM5NDM4Mi8zMDcyMjk4OTctZDQ2MjU4MDYtOThiOC00M2ZhLWFlMGQtY2Q0ODNlNzhjZTNiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA3VDAzMzQ0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJkNWU0MDBiMmZiODJjNTc0ZTIyMDZkNjBiNDUzNjAyZjkxNWQxM2IxMDQ2NTFjYjFhYmQ3ZTkzYWE1OTBhODEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Ol480__oD5796v5zQwh6uzoJ4F_2NvKSDaEg_5YWP0k)

Hello, I would like to ask if you only change the num_epoch in config.py from 15 to 5000 and maintain default values for other parameters? Look forward to your reply.

**Comment by HuangWanqiu** [comment]
> > I found the reason. It was my mistake. After increasing the epoch number to 5000 (actually around 3000 would suffice), the correlation on the GOD test set increased to around 0.57, as shown in the figure below.
> > Once again, thank you for your excellent work.
> > ![image](https://private-user-images.githubusercontent.com/12394382/307229897-d4625806-98b8-43fa-ae0d-cd483e78ce3b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3ODI3ODIsIm5iZiI6MTcwOTc4MjQ4MiwicGF0aCI6Ii8xMjM5NDM4Mi8zMDcyMjk4OTctZDQ2MjU4MDYtOThiOC00M2ZhLWFlMGQtY2Q0ODNlNzhjZTNiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA3VDAzMzQ0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJkNWU0MDBiMmZiODJjNTc0ZTIyMDZkNjBiNDUzNjAyZjkxNWQxM2IxMDQ2NTFjYjFhYmQ3ZTkzYWE1OTBhODEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Ol480__oD5796v5zQwh6uzoJ4F_2NvKSDaEg_5YWP0k)
> 
> Hello, I would like to ask if you only change the num_epoch in config.py from 15 to 5000 and maintain default values for other parameters? Look forward to your reply.

If I remember correctly, I only made very minor changes, such as adjusting the batch size and number of epochs.

**Comment by xyhanHIT** [comment]
> > > I found the reason. It was my mistake. After increasing the epoch number to 5000 (actually around 3000 would suffice), the correlation on the GOD test set increased to around 0.57, as shown in the figure below.
> > > Once again, thank you for your excellent work.
> > > ![image](https://private-user-images.githubusercontent.com/12394382/307229897-d4625806-98b8-43fa-ae0d-cd483e78ce3b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3ODI3ODIsIm5iZiI6MTcwOTc4MjQ4MiwicGF0aCI6Ii8xMjM5NDM4Mi8zMDcyMjk4OTctZDQ2MjU4MDYtOThiOC00M2ZhLWFlMGQtY2Q0ODNlNzhjZTNiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA3VDAzMzQ0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJkNWU0MDBiMmZiODJjNTc0ZTIyMDZkNjBiNDUzNjAyZjkxNWQxM2IxMDQ2NTFjYjFhYmQ3ZTkzYWE1OTBhODEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Ol480__oD5796v5zQwh6uzoJ4F_2NvKSDaEg_5YWP0k)
> > 
> > 
> > Hello, I would like to ask if you only change the num_epoch in config.py from 15 to 5000 and maintain default values for other parameters? Look forward to your reply.
> 
> If I remember correctly, I only made very minor changes, such as adjusting the batch size and number of epochs.

Okay, thank you for your reply！
